activity/journal.go

```diff
@@ -0,0 +1,67 @@
+package activity
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+)
+
+// Journal contains the command-line used to invoke the process associated with this item
+// and its journal ID.
+type Journal struct {
+	ID, Description string
+	Start           time.Time
+}
+
+var mux = sync.Mutex{}
+
+func initHistory() ([]Journal, map[string]struct{}) {
+	return make([]Journal, 0), make(map[string]struct{})
+}
+
+// history: a list of commands that have been invoked
+// jidToHistory: lookup to identify whether a journal ID has already been recorded in history.
+var history, jidToHistory = initHistory()
```

Comment:
`jidToHistory` => `storedJournals` -- latter better reflects how this variable is used since `struct{}` is not a history object.